cmd/srcd/cmd/sql_test.go

```diff
@@ -0,0 +1,215 @@
+// +build integration
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	cmdtest "github.com/src-d/engine/cmd/test-utils"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type SQLTestSuite struct {
+	cmdtest.IntegrationSuite
+	testDir string
+}
+
+func TestSQLTestSuite(t *testing.T) {
+	s := SQLTestSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *SQLTestSuite) SetupTest() {
+	var err error
+	s.testDir, err = ioutil.TempDir("", "sql-test")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (s *SQLTestSuite) TearDownTest() {
+	s.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+}
+
+const showTablesOutput = `+--------------+
+|    TABLE     |
++--------------+
+| blobs        |
+| commit_blobs |
+| commit_files |
+| commit_trees |
+| commits      |
+| files        |
+| ref_commits  |
+| refs         |
+| remotes      |
+| repositories |
+| tree_entries |
++--------------+
+`
+
+func (s *SQLTestSuite) TestInit() {
+	require := s.Require()
+
+	repoPath := filepath.Join(s.testDir, "reponame")
+	err := os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+
+	_, err = s.RunInit(context.TODO(), s.testDir)
+	require.NoError(err)
+
+	buf, err := s.RunSQL(context.TODO(), "select * from repositories")
+	require.NoError(err)
+
+	expected := `+---------------+
+| REPOSITORY ID |
++---------------+
+| reponame      |
++---------------+
+`
+	require.Contains(buf.String(), expected)
+}
+
+func (s *SQLTestSuite) TestNoInit() {
+	require := s.Require()
+
+	buf, err := s.RunSQL(context.TODO(), "show tables")
+	require.NoError(err)
+
+	require.Contains(buf.String(), showTablesOutput)
+}
+
+func (s *SQLTestSuite) TestValidQueries() {
+	testCases := []string{
+		"show tables",
+		"SHOW TABLES",
+		"show tables;",
+		"SHOW TABLES;",
+		" show tables ; ",
+		" SHOW TABLES ; ",
+		"	show tables	;	",
+		"	SHOW TABLES	;	",
+		"/* comment */ show tables;",
+		`/* multi line
+			comment */
+			show tables;`,
+	}
+
+	for _, query := range testCases {
+		s.T().Run(query, func(t *testing.T) {
+			assert := assert.New(t)
+
+			buf, err := s.RunSQL(context.TODO(), query)
+			assert.NoError(err)
+
+			assert.Contains(buf.String(), showTablesOutput)
+		})
+	}
+}
+
+func (s *SQLTestSuite) TestWrongQuery() {
+	testCases := []struct {
+		query string
+		err   string
+	}{
+		{
+			query: "show",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
```

Comment:
do we want to ensure that error is **exactly** that? even with error number and formating?

Reply:
what do you propose?
My thinking here is that we want to check that the error is in fact a syntax error, not some other random crash. For example if we just checked for an error, it could happen that our client crashes with a nil pointer panic when gitbase returns an error; and the test would pass.

Reply:
I agree that we must test that it is caused a syntax error, and not a random one; what I would suggest, is not caring about the error format. Maybe just searching for `syntax error at position` would be enough because the rest is added by different layers of Engine (which do not add extra value to the user), so imo it should not be validated in tests.

Reply:
I updated the test to check only the inner error, ignoring the `rcp error` prefix. Since it's a small change, I already squashed it.